File: Lab_1/public/backend/Square.php
<?php

require_once "Shape.php";

class Square extends Shape
{
  private float $x;
  private float $y;
  private bool $low_quadrant = FALSE;

  public function __construct(int $quadrant, float $x, float $y){
    parent::__construct($quadrant);
    $this->x = $x;
    $this->y = $y;
  }  

  public function isInShape(float $x, float $y): bool
  {
    if ($this->quadrant == 3 || $this->quadrant == 4) {
      $this->low_quadrant = TRUE;
    } 
    if ($this->low_quadrant)
    {
      if ($this->quadrant == 4)
      {
        return $this->isInQuadrant($x, $y) && ($this->x >= $x) && ($this->y <= $y);
      } else {
        return $this->isInQuadrant($x, $y) && ($this->x <= $x) && ($this->y <= $y);
      }
    } else {
      if ($this->quadrant == 1)
      {
        return $this->isInQuadrant($x, $y) && ($this->x >= $x) && ($this->y >= $y);
      } else {
        return $this->isInQuadrant($x, $y) && ($this->x <= $x) && ($this->y >= $y);
      }
    }
  }
}

File: Lab_2/src/main/java/Shapes/Square.java
package Shapes;
public class Square extends AbstractShape{

  private double x;
  private double y;
  private boolean low_quadrant = false;

  public Square( Integer quadrant, double x, double y ) {
    super(quadrant);
    this.x = x;
    this.y = y;
    if ( this.quadrant == 3 || this.quadrant == 4) { this.low_quadrant = true; } 
  }

  @Override
  public boolean isInShape(double x, double y) 
  { 
    if (this.low_quadrant)
    {
      if (this.quadrant == 4)
      {
        return this.isInQuadrant(x, y) && (this.x >= x) && (this.y <= y);
      } else {
        return this.isInQuadrant(x, y) && (this.x <= x) && (this.y <= y);
      }
    } else {
      if (this.quadrant == 1)
      {
        return this.isInQuadrant(x, y) && (this.x >= x) && (this.y >= y);
      } else {
        return this.isInQuadrant(x, y) && (this.x <= x) && (this.y >= y);
      }
    }
  }
  
}
